### PR TITLE
[Serve] Define BackendConfig protobuf and adapt it in Java

### LINKED ISSUE #16390: [serve] create protobuf definition for cross language calls
### Describe your feature request

For feature requests or questions, post on our Discussion page instead: https://discuss.ray.io/

### PR DESCRIPTION
**Pull request by liuyang-my**
## Why are these changes needed?

This PR is about Java users using Ray Serve. We defined the proto file of BackendConfig, so that Python Controller could startup Java Backend Actor.

Detailed design is in doc :https://docs.google.com/document/d/1b9VCZsC2jVhWw_c_ifMu_ED3NxkARcOnU-OFqUYu-r4/edit#heading=h.b3vvz7iwnixn

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/serve/src/main/java/io/ray/serve/RayServeWrappedReplica.java [hunk 1]
```diff
@@ -43,6 +53,24 @@ public RayServeWrappedReplica(
...
+      return new Object[0];
+    }
+
+    if (!backendConfig.getIsCrossLanguage()) {
+      // If the construction request is from Java API, deserialize initArgsbytes to Object[]
+      // directly.
+      Object result = Hessian2Seserializer.decode(initArgsbytes);
+      return (Object[]) result;
+    } else {
+      // TODO
```
Comment (simon-mo) [inline java/serve/src/main/java/io/ray/serve/RayServeWrappedReplica.java:69]: explain the TODO?

File: java/serve/src/main/java/io/ray/serve/serializer/Hessian2Seserializer.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package io.ray.serve.serializer;
+
+import com.caucho.hessian.io.Hessian2Input;
```
Comment (simon-mo) [inline java/serve/src/main/java/io/ray/serve/serializer/Hessian2Seserializer.java:3]: Why do we use Hessian as the serialization format here? If it's expected to be used cross language we should use messagepack, if not is Hessian a standard serialization protocol for Java? If so does Ray core use it?

**Review by simon-mo — APPROVED** [review]
LGTM overall, main question is about the use of Hessian as a serialization format for initArgs and user config.

File: java/serve/src/main/java/io/ray/serve/serializer/Hessian2Seserializer.java (hunk 2 above)
Comment (caitengwei) [reply java/serve/src/main/java/io/ray/serve/serializer/Hessian2Seserializer.java:3]: Hesssian's compatibility is better when adding new properties to the object. But MsgPack performance better in cross-language scenarios. I think MsgPack is the right choice here.

**Comment by simon-mo** [comment]
can you try merge master? the CI failure mostly come that environment issue

**Pull request merged**